### PR TITLE
Cache concept JSON to improve response time

### PR DESCRIPTION
**Pull request by kriskelly**
This caching code can/should probably be generalized to the other serializers, but I wanted to keep it local for the time being to keep from accidentally breaking things.

### CODE REVIEW

File: app/serializers/concept_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  delegate :cache_key, :to => :object
+
+  def to_json(*args)
+    Rails.cache.fetch expand_cache_key(self.class.to_s.underscore, cache_key, 'to-json') do
+      super
+    end
+  end
+
+  def parent_uid
+    object.parent.try(:uid)
```
Comment (marcellosachs) [inline app/serializers/concept_serializer.rb]: @kriskelly what do you think about just pulling this in with the SQL? Though I guess it depends on time constraints - this is probably fine for now.

File: app/serializers/concept_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class ConceptSerializer < ActiveModel::Serializer
+  attributes :id, :name, :uid, :parent_id, :level, :parent_uid
+  delegate :cache_key, :to => :object
+
+  def to_json(*args)
+    Rails.cache.fetch expand_cache_key(self.class.to_s.underscore, cache_key, 'to-json') do
```
Comment (marcellosachs) [inline app/serializers/concept_serializer.rb]: @kriskelly im not sure how urgent the 'response time' issue is, but perhaps we could also cache at the level of the entire query? ie if no concept has been added, deleted, or updated, then use cache for the whole response

Comment (marcellosachs) [inline app/serializers/concept_serializer.rb]: @kriskelly if we are able to pull in the parent_uid with SQL, then a cache at the individual level would no longer be necessary

Comment (kriskelly) [inline app/serializers/concept_serializer.rb]: @marcellosachs The SQL execution time isn't the problem, it's actually rendering the whole thing to JSON that seems to take awhile. At least that's what I get from the logs. This is not the real example, but it's something like this, where the `Views` time is much, much higher than `ActiveRecord`.

```
Completed 200 OK in 788ms (Views: 741.6ms | ActiveRecord: 4.2ms)
```

You can try it for yourself, try firing up the postgres console on staging and running the query that gets generated by Concept.all_with_level. It's very fast.

**Comment by kriskelly** [comment]
@marcellosachs This is actually kind of urgent because the default load time for that query is somewhere in the range of 6 - 8 seconds, which will happen every time someone loads the administrative CMS for Quill Grammar.

**Comment by marcellosachs** [comment]
@kriskelly ok makes sense!

**Pull request merged**